examples/carousel/ui/carousel.slint

```diff
@@ -48,29 +49,47 @@ export component Carousel inherits FocusScope {
         }
     }
 
-    TouchArea {
-        clicked => {
-            focus-scope.focus()
-        }
-
-        width: parent.width;
-        height: parent.height;
-     }
-
-    Rectangle {
-        clip: true;
-        background: transparent;
+    swipe := SwipeGestureRecognizer {
+        swipe-left: true;
+        swipe-right: true;
 
-        Flickable {
-            interactive: false;
-            viewport-x: root.center-x - root.selected-index * (root.itemWidth + root.spacing);
-
-            animate viewport-x { duration: root.duration; easing: ease-in; }
+        swiped => {
+            if self.current-position.x > self.pressed-position.x + root.itemWidth / 2 {
+                root.move-left();
+            } else if self.current-position.x < self.pressed-position.x - root.itemWidth / 2 {
+                root.move-right();
+            }
```

Comment:
Ahh, I see now why parameters to `swiped` don't always help. An application specific swipe threshold makes sense to apply.